### ISSUE TITLE
Nonnumeric browser versions + allowHigherVersions causes all versions to be in the range

### ISSUE THREAD

**Issue by rayproud-pp** [browserslist/browserslist-useragent-regexp#872]
Hey!

We've discovered an interesting issue. 
When we use `unreleased versions` in our `.browserslistrc` with `allowHigherVersions` in the `browserslist-useragent-regexp` setting, the result regex matches all versions of Safari.

After some debugging, we found out that having `unreleased versions` in `.browserslistrc` results in having `Safari TP` among Safari versions. Which can't be handled by the [semverify](https://github.com/browserslist/browserslist-useragent-regexp/blob/ab4200b92a1cda8f909ec8e66bb7f8e68d1af18b/src/semver/semver.ts#L23) function. There's even a comment says "risky" there 😅

Anyway, "Safari TP" turns into `0.0.0`, which, with the `allowHigherVersions` flag, makes the whole browserslist setting useless — the result regex included everything that's greater than 0.
![Screenshot 2021-09-01 at 14 55 11](https://user-images.githubusercontent.com/61588580/131694991-11de1c87-22a8-4632-ba76-6d5094c16bb9.png)

This is the result regex:
`(Version\/(0|([1-9]|\d{2,})|12|(1[3-9]|[2-9]\d|\d{3,})|13|(1[4-9]|[2-9]\d|\d{3,})|14|(1[5-9]|[2-9]\d|\d{3,})|15|(1[6-9]|[2-9]\d|\d{3,}))\.\d+(?:\.\d+)? Safari\/)`

instead of awaited
`(Version\/(12|(1[3-9]|[2-9]\d|\d{3,})|13|(1[4-9]|[2-9]\d|\d{3,})|14|(1[5-9]|[2-9]\d|\d{3,})|15|(1[6-9]|[2-9]\d|\d{3,}))\.\d+(?:\.\d+)? Safari\/)`.

**Data for debugging:**
`.browserslistrc` for testing:
```
unreleased versions
last 2 years
last 3 major versions
not dead
not ie <= 11
not ie_mob <= 11
```

Version of `browserslist-useragent-regexp` — `3.0.0`
Version of `node` — `14.17`

**Comment by rayproud-pp** [comment]
Hey @dangreen!
Will you have time next week to take a look at the issue?

If not, I can help with a PR, just drop me a line 🙂

**Comment by dangreen** [comment]
@rayproud-pp Hi! Yes, I will get it soon.

**Comment by rayproud-pp** [comment]
@dangreen, any updates on this issue?

**Comment by dangreen** [comment]
Fixed in [v3.0.1](https://www.npmjs.com/package/browserslist-useragent-regexp/v/3.0.1)

**Comment by dangreen** [comment]
@rayproud-pp Hi. Did you test fix?
